kdc/kerberos5.c

```diff
@@ -955,7 +958,7 @@ static const struct kdc_patypes pat[] = {
     { KRB5_PADATA_PA_PK_OCSP_RESPONSE , "OCSP", 0, NULL },
     { 
 	KRB5_PADATA_ENC_TIMESTAMP , "ENC-TS",
-	PA_ANNOUNCE,
+	PA_ANNOUNCE | PA_NOT_IN_FAST | PA_DONT_ANNOUNCE_FAST,
```

Comment:
This is the opposite of what we should want.  We should want `KRB5_PADATA_ENC_TIMESTAMP` to only be used inside a FAST armored tunnel.

But more importantly, it's pointless.  What matters as far as not sending encrypted timestamps not super-encrypted is the _client_.  If a `kinit` can be spoofed into sending an encrypted timestamp outside a FAST armored tunnel, then the password is compromised (because it can be attacked offline at that point).

Reply:
I'm inclined to accept (I'm even coding, as we speak) a patch to add two configuration parameters:

- `[kdc] enable_enc_timestamp = BOOL`
- `[kdc] require_armored_weak_preauth = BOOL`

and if `require_armored_weak_preauth = true` then allow enc-timestamp only inside armored FAST and only advertise it inside armored FAST.

Reply:
I'm also thinking of a HDB principal entry flag for "no enc timestamp".

Reply:
We should patch our kinit to always try `pk_anon_fast_armor` unless an armor ccache is given.

Reply:
KRB5_PADATA_ENC_TIMESTAMP only uses the client's key, not the armor key, for validation, so don't we want to disallow it with FAST? Shouldn't it rather be KRB5_PADATA_ENCRYPTED_CHALLENGE that is only used inside a FAST tunnel?

Reply:
Oh, no, I'm confused.  `ENC-CHAL` replaces `ENC-TIMESTAMP` inside FAST.